RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,200 @@
+
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love, Soneson, Hickey, and Patro.
+
+We'll need the `quant.sf` files for all the samples in an experiment.
+
+## Libraries and functions
+
+```{r library, live = TRUE}
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+```{r directories, live = TRUE}
+# directory where the data are located
+data_dir <- file.path("data", "gastric-cancer")
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "gastric-cancer_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names, live = TRUE}
+# Name the output gastric-cancer_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "gastric-cancer_tximeta.RDS")
+```
+
+### Naming
+
+All output files from `salmon quant` we'll use with `tximeta` are named 
+`quant.sf`.
+The file names do not have any information about the sample name.
+
+```{r sf_files, live = TRUE}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` 
+package.
+
+Notice how the file path is separated by `/`.
+If we were to split up this character by `/`, the fourth item is the sample 
+name.
+This is exactly what `stringr::word` allows us to do: split up the file paths
+by `/` and extract the sample names.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, 4, sep = "/")
+sample_names
+```
+
+`tximeta` needs a data.frame with at least these two columns: 
+- a `files` column  with the file paths to the quant.sf files
+- a `names` column with the sample names
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names, 
+                      stringsAsFactors = FALSE)
+```
+
+### Set up metadata
+
+And now we'll read in the sample metadata.
+
+```{r sample_meta_df, live = TRUE}
+# Read in the sample metadata TSV file and have a look
+sample_meta_df <- readr::read_tsv(meta_file)
+sample_meta_df
+```
+
+We'll want this information to be added to the `coldata`. 
```

Comment:
I don't think we will have covered joins at this point, so we can either follow the previous model of checking that samples are in the same order and appending, or we probably need a more extensive description of what is happening here.

What is the metadata that we have? How is it arranged? (We have them look, but we should discuss the content, at least with a questioning prompt)

Reply:
We cover joins in intro-to-r: https://github.com/AlexsLemonade/training-modules/blob/ebb928ed6499cc7ae41854e470998f75c4757fe4/intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd#L493

Reply:
Yay past us!